source/includes/_account.md

```diff
@@ -1022,7 +1022,7 @@ If successful, returns a status code of 200 and a list of all accounts.
 
 ### Search accounts
 
-Search for an account by a specified search string. The search string is compared to the following attributes: `accountId`, `name`, `email`, `companyName`, and `externalKey`. The operation returns all account records in which the search string matches all or part of any one of these attributes. Note that this endpoint returns a shallow object (and thus the value `0` is returned for fields like `accountBalance`). In order to retrieve the actual data for such fields, additional endpoints (like retrieve account by id) would need to be invoked.
+Search for an account by a specified search string. The search string is compared to the following attributes: `accountId`, `name`, `email`, `companyName`, and `externalKey`. The operation returns the account record in which the search string matches any one of these attributes. Note: the string must match the entire attribute, not just a part of it.
```

Comment:
I thought we had recently added the fact that 'shallow object' are being returned, so why are we removing this?

Also, regarding the note, I think this is slightly misleading: It is true that it must match the entire attribute but at the same time, it does not need to be an exact search as shown [here](https://github.com/killbill/killbill/blob/master/account/src/main/resources/org/killbill/billing/account/dao/AccountSqlDao.sql.stg#L118). Perhaps showing a few examples would be useful? @reshmabidikar Any thoughts?

Reply:
> I thought we had recently added the fact that 'shallow object' are being returned, so why are we removing this?

I think this is to do with [this comment](https://github.com/killbill/slate/pull/171#issue-1313615932). So basically, the `accountWithBalance` and `accountWithBalanceAndCBA` flags control whether the balance is returned or not which is why I think @KBbitsP has made this change. Can you confirm @KBbitsP ?

Reply:
> Also, regarding the note, I think this is slightly misleading: 

Yes, I agree. @KBbitsP, the previous text read "The operation returns all account records in which the search string matches all or part of any one of these attributes". Is there any particular reason for removing this text? That is, did you observe any behavior that contradicts this? If not, I think we should just stick to this text and provide some examples as @sbrossie suggested.

Reply:
Hi @reshmabidikar @sbrossie 

Yes I checked this again. 

```
Example 1 : http://localhost:8080/1.0/kb/accounts/search/26c5aaa8-81dd-4ee7-959b-24f53b42267a?offset=0&limit=100&accountWithBalance=false&accountWithBalanceAndCBA=false&audit=NONE

{
    "accountId": "26c5aaa8-81dd-4ee7-959b-24f53b42267a",
    "name": "20thAugCus8FN 20thAugCus8LN",
    "firstNameLength": 13,
    "externalKey": "20thAugCus31Days",
    "email": null,
    "billCycleDayLocal": 1,
    "currency": "EUR",
    "parentAccountId": null,
    "isPaymentDelegatedToParent": false,
    "paymentMethodId": null,
    "referenceTime": "2022-08-20T17:25:34.000Z",
    "timeZone": "Etc/UTC",
    "address1": null,
    "address2": null,
    "postalCode": null,
    "company": null,
    "city": null,
    "state": null,
    "country": "FR",
    "locale": "en_US",
    "phone": null,
    "notes": null,
    "isMigrated": false,
    "accountBalance": null,
    "accountCBA": null,
    "auditLogs": []
  }
```


```
Example 2 : http://localhost:8080/1.0/kb/accounts/search/26c5aaa8-81dd-4ee7-959b-24f53b42267a?offset=0&limit=100&accountWithBalance=true&accountWithBalanceAndCBA=true&audit=NONE

{
    "accountId": "26c5aaa8-81dd-4ee7-959b-24f53b42267a",
    "name": "20thAugCus8FN 20thAugCus8LN",
    "firstNameLength": 13,
    "externalKey": "20thAugCus31Days",
    "email": null,
    "billCycleDayLocal": 1,
    "currency": "EUR",
    "parentAccountId": null,
    "isPaymentDelegatedToParent": false,
    "paymentMethodId": null,
    "referenceTime": "2022-08-20T17:25:34.000Z",
    "timeZone": "Etc/UTC",
    "address1": null,
    "address2": null,
    "postalCode": null,
    "company": null,
    "city": null,
    "state": null,
    "country": "FR",
    "locale": "en_US",
    "phone": null,
    "notes": null,
    "isMigrated": false,
    "accountBalance": 1360,
    "accountCBA": 0,
    "auditLogs": []
  }
```

So balance depends on the query params.

Reply:
Hi @reshmabidikar 

Yes, these are the search strings allowed : accountId, name, email, companyName, and externalKey

In case of accountID an exact match is needed whereas for others, a part match will do.

Reply:
okay, could you please update the docs for this? So, maybe use the original text but mention that a full match is required for `accountId` but a partial match works for the other fields? Also, add a few examples?